url/url.go

```
@@ -0,0 +1,339 @@
+//
+//	To add query parameters to a URL:
+//	  params := map[string]string{"key": "value", "foo": "bar"}
+//	  updatedURL, err := AddQueryParams("http://example.com", params)
+//	  fmt.Println("Result:", updatedURL)
+func AddQueryParams(urlStr string, params map[string]string) (string, error) {
+	parsedURL, err := url.Parse(urlStr)
+	if err != nil {
+		return "", errors.New("URL could not be parsed")
+	}
+	if urlStr != "" {
+		re := regexp.MustCompile("^(https?|wss?|ftp):\\/\\/[a-zA-Z0-9-]+(\\.[a-zA-Z0-9-]+)+(\\?[a-zA-Z0-9&=_-]*)?$")
```

golangci-lint

S1007: should use raw string (`...`) with regexp.MustCompile to avoid having to escape twice

(gosimple)